### PR TITLE
Azure DevOps - a new remote type

### PR DESCRIPTION
**Pull request by legendary-mich**

### CODE REVIEW

File: browse-at-remote.el [hunk 1]
```diff
@@ -253,6 +255,47 @@ If HEAD is detached, return nil."
...
+   (linestart (format "%s&line=%d&lineStartColumn=1&lineEndColumn=1" base-url linestart))
+   (t base-url))
+  )
+  )
+
+(defun browse-at-remote--format-commit-url-as-ado (repo-url commithash)
+  "Commit URL formatted for ado"
+  ;; They does not seem to have anything like permalinks from github.
+  (error "The ado version of the commit-url is not implemented")
+  ;; (format "%s/commits/%s" repo-url commithash))
```
Comment (rmuslimov) [inline browse-at-remote.el:296]: nit: pls remove if is not needed

File: browse-at-remote.el [hunk 2]
```diff
@@ -253,6 +255,47 @@ If HEAD is detached, return nil."
...
+                           (browse-at-remote-ado-format-url repo-url)
+                           branch-prefix
+                           location
+                           filename))
+         )
+  (cond
+   ((and linestart lineend)
+    (format "%s&line=%d&lineEnd=%d&lineStartColumn=1&lineEndColumn=1" base-url linestart (+ 1 lineend)))
+   (linestart (format "%s&line=%d&lineStartColumn=1&lineEndColumn=1" base-url linestart))
+   (t base-url))
```
Comment (rmuslimov) [inline browse-at-remote.el:288]: pls put closing parentheses on same line, refer https://github.com/bbatsov/emacs-lisp-style-guide

File: browse-at-remote.el [hunk 3]
```diff
@@ -253,6 +255,47 @@ If HEAD is detached, return nil."
...
+    )
+  )
+
+(defun browse-at-remote--format-region-url-as-ado (repo-url location filename &optional linestart lineend)
+  "URL formatted for ado"
+  (let* (
+         ;; NOTE: I'm not sure what's the meaning of the
+         ;; branch-prefix. My guess is that it stands for a "Git
+         ;; Branch".
+         (branch-prefix "GB")
```
Comment (rmuslimov) [inline browse-at-remote.el:277]: may be better if we put this just inside template, no need to extract if we're not using it

**Review by rmuslimov — COMMENTED** [review]
overall, lgtm. Please address comments and we can merge. Thanks for adding!

File: browse-at-remote.el (hunk 1 above)
Comment (legendary-mich) [reply browse-at-remote.el]: Done.

File: browse-at-remote.el (hunk 2 above)
Comment (legendary-mich) [reply browse-at-remote.el]: Done, though I hate that convention. It makes hard to move the last line up, or copy it and paste in another place.

File: browse-at-remote.el (hunk 3 above)
Comment (legendary-mich) [reply browse-at-remote.el]: Done.

**Comment by legendary-mich** [comment]
@rmuslimov  I've fixed all the style issues pointed by you. Please, review.

**Comment by rmuslimov** [comment]
Sounds great, thanks for addressing comments

**Pull request merged**